airbyte-workers/src/main/java/io/airbyte/workers/DbtTransformationRunner.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.workers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.io.LineGobbler;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.config.OperatorDbt;
+import io.airbyte.workers.normalization.NormalizationRunner;
+import io.airbyte.workers.process.ProcessBuilderFactory;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.tools.ant.types.Commandline;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DbtTransformationRunner implements AutoCloseable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DbtTransformationRunner.class);
+
+  private final ProcessBuilderFactory pbf;
+  private final NormalizationRunner normalizationRunner;
+  private Process process = null;
+
+  public DbtTransformationRunner(final ProcessBuilderFactory pbf, NormalizationRunner normalizationRunner) {
+    this.pbf = pbf;
+    this.normalizationRunner = normalizationRunner;
+  }
+
+  public void start() throws Exception {
+    normalizationRunner.start();
+  }
+
+  /**
+   * Re-use the Normalization runner to configure the dbt project with the correct destination
+   * settings and then run the custom transformation command.
+   */
+  public boolean run(String jobId, int attempt, Path jobRoot, JsonNode config, OperatorDbt dbtConfig) throws Exception {
+    if (!normalizationRunner.configureDbt(jobId, attempt, jobRoot, config, dbtConfig)) {
```

Comment:
unless we think the DbtTransformationRunner will rely on NormalizationRunner in the future, I feel pretty strongly that we should try and pull the shared configuration function out instead of injecting the NormalizationRunner into the DbtTransformationRunner. We would be able to get rid of the implicit relationship we now have.

Reply:
As we don't have any control on the content (what software is installed and where? does it have a python env? which version?) of the docker image for the dbt transformations (as they are specified by the user), I am using the normalization docker image where we install our own python scripts (with its dependencies). 

So i can use these scripts to generate valid dbt profiles.yml files and pull code from git repository to build the workspace to run the userś docker image from.

Reply:
makes sense! I didn't understand everything when I reviewed this. your decision makes sense. thanks for explaining.